### PR TITLE
Add test for push/pop operators + Fix a slew of PC related issues

### PR DESCRIPTION
**Pull request by dezuman**

### CODE REVIEW

File: src/narmvm.rs [hunk 1]
```diff
@@ -105,8 +105,10 @@ impl NarmVM{
                 //25 bits total length
                 let imm32 = sign_extend32(value, 25);
                 let lr = LongRegister{register: 14};
-                self.set_reg(&lr, self.virtual_pc | (self.pc & 1)); //or with bottom bit of current pc to copy interworking mode
+                
+                self.set_reg(&lr, self.virtual_pc | 1); //or with bottom bit of current pc to copy interworking mode
```
Comment (Earlz) [inline src/narmvm.rs]: self.pc & 1 is the more proper way to do this, but this is fine since we aren't planning to go beyond thumb mode anytime soon

File: src/narmvm.rs [hunk 2]
```diff
@@ -970,10 +972,12 @@ impl NarmVM{
...
             return msg;
         }
-        msg.push_str("Execution flow since breakpoint: \n");
+        msg.push_str("\nExecution flow since breakpoint: \n");
+        msg.push_str("\n[pc hex value] -- [pc bytes after entry] -- op binary value -- op hex value\n\n");
         for (pc, op) in &self.executed_opcodes{
-            msg.push_str(&format!("[{:#010x}] -- {} -- {:#06x}\n",
+            msg.push_str(&format!("[{:#08x}] -- [{:#04}] -- {} -- {:#06x}\n",
                 pc,
+                pc - 0x01_0000,
```
Comment (Earlz) [inline src/narmvm.rs:980]: Why is this needed?

**Review by Earlz — APPROVED** [review]

File: src/narmvm.rs (hunk 2 above)
Comment (dezuman) [reply src/narmvm.rs:980]: Oh, figured it would be helpful for tracing larger programs to calculate the byte offset from program start and show as decimal.

File: src/narmvm.rs (hunk 1 above)
Comment (dezuman) [reply src/narmvm.rs]: Fixed it
